docs/pages/guides/nextjs-starter-kit.mdx

```diff
@@ -0,0 +1,545 @@
+import from the `/lib/client` folder, and omit `res` and `res` instead. This
+works by calling an API endpoint for you and then calling the `/lib/server`
+function.
+
+```tsx
+import { createDocument } from "../../lib/client";
+
+export function CreateDocumentButton() {
+  async function handleCreateDocument() {
+    // Create a new document
+    const { data, error } = await createDocument({
+      name: "My document",
+      type: "whiteboard",
+      userId: "charlie.layne@example.com",
+    });
+    // ...
+  }
+  return <button onClick={handleCreateDocument}>New document</button>;
+}
+```
+
+Functions that return data can be used with [SWR hooks](https://swr.vercel.app/)
+that automatically update your data in components. For example,
+`getDocumentUsers` returns a list of users with access to the room:
+
+```tsx
+// Convert from this
+const { data, error } = await getDocumentUsers({
+  documentId: "my-document-id",
+});
+
+// To this
+const { data, error } = useDocumentsFunctionSWR([
+  getDocumentUsers,
+  {
+    documentId: "my-document-id",
+  },
+]);
+```
+
+Here's a working example:
+
+```tsx
+import { getDocumentUsers, useDocumentsFunctionSWR } from "../../lib/client";
+
+export function ListUsers() {
+  // Get users attached to a document and update every 1000ms
+  const { data: users, error: usersError } = useDocumentsFunctionSWR(
+    [
+      getDocumentUsers,
+      {
+        documentId: "my-document-id",
+      },
+    ],
+    { refreshInterval: 1000 }
+  );
+
+  if (usersError) {
+    return <div>Error</div>;
+  }
+
+  if (!users) {
+    return <div>Loading...</div>;
+  }
+
+  return (
+    <ul>
+      {users.map((user) => (
+        <li>user.name</li>
+      ))}
+    </ul>
+  );
+}
+```
+
+### How to extend the `Document` type
+
+If you'd like to add a new property to `Document`, it's simple. First, edit the
+`Document` type in `/types/documents.ts`:
+
+```ts file="types/documents.ts"
+export type Document = {
+  // Your new property
+  randomNumber: number;
+  //...
+};
+```
+
+Then modify the return value in `/lib/server/utils/buildDocuments.ts`. This is a
+function that converts a Liveblocks room into your custom document format:
+
+```ts file="lib/server/utils/buildDocuments.ts"
+// Return our custom Document format
+const document: Document = {
+  randomNumber: Math.random(),
+  // ...
+};
+```
+
+Next, run the following command to check for problems:
+
+```
+npm run typecheck
+```
+
+If no errors are returned, the document properties were successfully extended.
+
+### How to extend the `User` & `Session` type
+
+Similar to the way we extend the `Document` type, we can also extend the `User`
+and `Session` type.
+
+Adding a new property to `User`/`Session` is simple. First, edit the `User` type
+in `/types/data.ts`.
+
+```ts file="types/data.ts"
+export type User = {
+  // Your new property
+  randomNumber: number;
+  // ...
+};
+```
+
+Then make sure to return this new property in `/lib/server/database/getUser.ts`.
```

Comment:
```suggestion
Then make sure to return this new property in `[/lib/server/database/getUser.ts](https://github.com/liveblocks/nextjs-starter-kit/blob/main/lib/server/database/getUser.ts)`.
```

Every time we refer to a file, I was wondering if it wouldn’t be cool to link it directly to the file in OUR repo? So that people browsing through the guide that haven’t installed the starter kit yet but are evaluating can quickly  see what we're referring to

_(We'd have to change the repo compared to what I'm suggesting)_